### PR TITLE
[DA-110] Create REST API for automated BC creation

### PR DESCRIPTION
**Pull request by jomrazek**

### CODE REVIEW

File: bc-rest/src/main/java/org/jboss/da/bc/BuildConfigurationsProduct.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    @Path("/finish-process")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Finish analyse and create BCs", response = FinishResponse.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Response succesfully generated") })
+    public FinishResponse finishAnalyse(@ApiParam(
+            value = "Complete detail information needed to create BCs") InfoEntity bc) {
+        return getFinishAnalyseResponse();
+    }
+
+    // TODO remove after implementation
```
Comment (thescouser89) [inline bc-rest/src/main/java/org/jboss/da/bc/BuildConfigurationsProduct.java]: Should this method still be here? :smiley:

File: bc-rest/src/main/java/org/jboss/da/bc/model/BuildConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package org.jboss.da.bc.model;
```
Comment (thescouser89) [inline bc-rest/src/main/java/org/jboss/da/bc/model/BuildConfiguration.java]: You could perhaps re-use the model from the `communication` module.
```
communication/src/main/java/org/jboss/da/communication/pnc/model/BuildConfiguration.java
```

File: bc-rest/src/main/java/org/jboss/da/bc/BuildConfigurationsProduct.java (hunk 1 above)
Comment (jbartece) [inline bc-rest/src/main/java/org/jboss/da/bc/BuildConfigurationsProduct.java]: Yes :-). It is a Mock of data, which will be there until we finish this feature :-)

File: bc-rest/src/main/java/org/jboss/da/bc/model/BuildConfiguration.java (hunk 2 above)
Comment (jbartece) [inline bc-rest/src/main/java/org/jboss/da/bc/model/BuildConfiguration.java]: These entities are different. BC in communication is PNC BC, but this BC is our model for BC import process in PNC. The entities are slightly different

**Comment by jbartece** [comment]
:+1:

**Comment by janinko** [comment]
:+1:

**Pull request merged**